<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.8"/>
<title>OpenMM: Modeller Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
  $(window).load(resizeHeight);
</script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">OpenMM
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.8 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="hierarchy.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html','');});
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Pages</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="summary">
<a href="#nested-classes">Classes</a> &#124;
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-static-methods">Static Public Member Functions</a> &#124;
<a href="#pub-attribs">Public Attributes</a> &#124;
<a href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">Modeller Class Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html" title="Modeller provides tools for editing molecular models, such as adding water or missing hydrogens...">Modeller</a> provides tools for editing molecular models, such as adding water or missing hydrogens.  
 <a href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#details">More...</a></p>
<div id="dynsection-0" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-0-trigger" src="closed.png" alt="+"/> Inheritance diagram for Modeller:</div>
<div id="dynsection-0-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-0-content" class="dyncontent" style="display:none;">
 <div class="center">
  <img src="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.png" usemap="#Modeller_map" alt=""/>
  <map id="Modeller_map" name="Modeller_map">
</map>
 </div></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:ac775ee34451fdfa742b318538164070e"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ac775ee34451fdfa742b318538164070e">__init__</a></td></tr>
<tr class="memdesc:ac775ee34451fdfa742b318538164070e"><td class="mdescLeft">&#160;</td><td class="mdescRight">Create a new <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html" title="Modeller provides tools for editing molecular models, such as adding water or missing hydrogens...">Modeller</a> object.  <a href="#ac775ee34451fdfa742b318538164070e">More...</a><br /></td></tr>
<tr class="separator:ac775ee34451fdfa742b318538164070e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6bbbabdcb078f5feb40e195428ee93ee"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a6bbbabdcb078f5feb40e195428ee93ee">getTopology</a></td></tr>
<tr class="memdesc:a6bbbabdcb078f5feb40e195428ee93ee"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the Topology of the model.  <a href="#a6bbbabdcb078f5feb40e195428ee93ee">More...</a><br /></td></tr>
<tr class="separator:a6bbbabdcb078f5feb40e195428ee93ee"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a97b90d5998ed9203e18087203e3eef2a"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a97b90d5998ed9203e18087203e3eef2a">getPositions</a></td></tr>
<tr class="memdesc:a97b90d5998ed9203e18087203e3eef2a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the atomic positions.  <a href="#a97b90d5998ed9203e18087203e3eef2a">More...</a><br /></td></tr>
<tr class="separator:a97b90d5998ed9203e18087203e3eef2a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a66aa7c8063db6217a0a0061f8b7ba206"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a66aa7c8063db6217a0a0061f8b7ba206">add</a></td></tr>
<tr class="memdesc:a66aa7c8063db6217a0a0061f8b7ba206"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add chains, residues, atoms, and bonds to the model.  <a href="#a66aa7c8063db6217a0a0061f8b7ba206">More...</a><br /></td></tr>
<tr class="separator:a66aa7c8063db6217a0a0061f8b7ba206"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab48bde70a0927a83d62464cb0627c34b"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ab48bde70a0927a83d62464cb0627c34b">delete</a></td></tr>
<tr class="memdesc:ab48bde70a0927a83d62464cb0627c34b"><td class="mdescLeft">&#160;</td><td class="mdescRight">Delete chains, residues, atoms, and bonds from the model.  <a href="#ab48bde70a0927a83d62464cb0627c34b">More...</a><br /></td></tr>
<tr class="separator:ab48bde70a0927a83d62464cb0627c34b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acb8683faf05f3bad2e7fd3985bf64963"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#acb8683faf05f3bad2e7fd3985bf64963">deleteWater</a></td></tr>
<tr class="memdesc:acb8683faf05f3bad2e7fd3985bf64963"><td class="mdescLeft">&#160;</td><td class="mdescRight">Delete all water molecules from the model.  <a href="#acb8683faf05f3bad2e7fd3985bf64963">More...</a><br /></td></tr>
<tr class="separator:acb8683faf05f3bad2e7fd3985bf64963"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6378af56ae75acce07d8a2f08f8cf87c"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a6378af56ae75acce07d8a2f08f8cf87c">convertWater</a></td></tr>
<tr class="memdesc:a6378af56ae75acce07d8a2f08f8cf87c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Convert all water molecules to a different water model.  <a href="#a6378af56ae75acce07d8a2f08f8cf87c">More...</a><br /></td></tr>
<tr class="separator:a6378af56ae75acce07d8a2f08f8cf87c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad489382394ab7f160e007ed890179435"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ad489382394ab7f160e007ed890179435">addSolvent</a></td></tr>
<tr class="memdesc:ad489382394ab7f160e007ed890179435"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add solvent (both water and ions) to the model to fill a rectangular box.  <a href="#ad489382394ab7f160e007ed890179435">More...</a><br /></td></tr>
<tr class="separator:ad489382394ab7f160e007ed890179435"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab19e4d48d3e7a317038fa32b82702ba6"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ab19e4d48d3e7a317038fa32b82702ba6">addHydrogens</a></td></tr>
<tr class="memdesc:ab19e4d48d3e7a317038fa32b82702ba6"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add missing hydrogens to the model.  <a href="#ab19e4d48d3e7a317038fa32b82702ba6">More...</a><br /></td></tr>
<tr class="separator:ab19e4d48d3e7a317038fa32b82702ba6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad59ef5b703f4dd7511858aa5d0be359d"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ad59ef5b703f4dd7511858aa5d0be359d">addExtraParticles</a></td></tr>
<tr class="memdesc:ad59ef5b703f4dd7511858aa5d0be359d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add missing extra particles to the model that are required by a force field.  <a href="#ad59ef5b703f4dd7511858aa5d0be359d">More...</a><br /></td></tr>
<tr class="separator:ad59ef5b703f4dd7511858aa5d0be359d"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-static-methods"></a>
Static Public Member Functions</h2></td></tr>
<tr class="memitem:a4d38bb19ab89760311eb99247c214948"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a4d38bb19ab89760311eb99247c214948">loadHydrogenDefinitions</a></td></tr>
<tr class="memdesc:a4d38bb19ab89760311eb99247c214948"><td class="mdescLeft">&#160;</td><td class="mdescRight">Load an XML file containing definitions of hydrogens that should be added by <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ab19e4d48d3e7a317038fa32b82702ba6" title="Add missing hydrogens to the model. ">addHydrogens()</a>.  <a href="#a4d38bb19ab89760311eb99247c214948">More...</a><br /></td></tr>
<tr class="separator:a4d38bb19ab89760311eb99247c214948"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-attribs"></a>
Public Attributes</h2></td></tr>
<tr class="memitem:a5e8a38af13069a0cc3fff9aae26892e4"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a5e8a38af13069a0cc3fff9aae26892e4">topology</a></td></tr>
<tr class="memdesc:a5e8a38af13069a0cc3fff9aae26892e4"><td class="mdescLeft">&#160;</td><td class="mdescRight">The Topology describing the structure of the system.  <a href="#a5e8a38af13069a0cc3fff9aae26892e4">More...</a><br /></td></tr>
<tr class="separator:a5e8a38af13069a0cc3fff9aae26892e4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac17d0e3d420545b0cbe7c718cf026ddc"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ac17d0e3d420545b0cbe7c718cf026ddc">positions</a></td></tr>
<tr class="memdesc:ac17d0e3d420545b0cbe7c718cf026ddc"><td class="mdescLeft">&#160;</td><td class="mdescRight">The list of atom positions.  <a href="#ac17d0e3d420545b0cbe7c718cf026ddc">More...</a><br /></td></tr>
<tr class="separator:ac17d0e3d420545b0cbe7c718cf026ddc"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html" title="Modeller provides tools for editing molecular models, such as adding water or missing hydrogens...">Modeller</a> provides tools for editing molecular models, such as adding water or missing hydrogens. </p>
<p>To use it, create a <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html" title="Modeller provides tools for editing molecular models, such as adding water or missing hydrogens...">Modeller</a> object, specifying the initial Topology and atom positions. You can then call various methods to change the model in different ways. Each time you do, a new Topology and list of coordinates is created to represent the changed model. Finally, call <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a6bbbabdcb078f5feb40e195428ee93ee" title="Get the Topology of the model. ">getTopology()</a> and <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a97b90d5998ed9203e18087203e3eef2a" title="Get the atomic positions. ">getPositions()</a> to get the results. </p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="ac775ee34451fdfa742b318538164070e"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def __init__ </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>topology</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>positions</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Create a new <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html" title="Modeller provides tools for editing molecular models, such as adding water or missing hydrogens...">Modeller</a> object. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">topology</td><td>(Topology) the initial Topology of the model </td></tr>
    <tr><td class="paramname">positions</td><td>(list) the initial atomic positions </td></tr>
  </table>
  </dd>
</dl>

<p>Referenced by <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ad489382394ab7f160e007ed890179435">Modeller.addSolvent()</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="a66aa7c8063db6217a0a0061f8b7ba206"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def add </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>addTopology</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>addPositions</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Add chains, residues, atoms, and bonds to the model. </p>
<p>Specify what to add by providing a new Topology object and the corresponding atomic positions. All chains, residues, atoms, and bonds contained in the Topology are added to the model.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">addTopoology</td><td>(Topology) a Topology whose contents should be added to the model </td></tr>
    <tr><td class="paramname">addPositions</td><td>(list) the positions of the atoms to add </td></tr>
  </table>
  </dd>
</dl>

<p>References <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmCrdFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmCrdFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ac17d0e3d420545b0cbe7c718cf026ddc">Modeller.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1desmonddmsfile_1_1DesmondDMSFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">DesmondDMSFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmRstFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmRstFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacsgrofile_1_1GromacsGroFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">GromacsGroFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a5e8a38af13069a0cc3fff9aae26892e4">Modeller.topology</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1amberprmtopfile_1_1AmberPrmtopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">AmberPrmtopFile.topology</a>, and <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacstopfile_1_1GromacsTopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">GromacsTopFile.topology</a>.</p>

<p>Referenced by <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ad59ef5b703f4dd7511858aa5d0be359d">Modeller.addExtraParticles()</a>.</p>

</div>
</div>
<a class="anchor" id="ad59ef5b703f4dd7511858aa5d0be359d"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def addExtraParticles </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>forcefield</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Add missing extra particles to the model that are required by a force field. </p>
<p>Some force fields use "extra particles" that do not represent actual atoms, but still need to be included in the System. Examples include lone pairs, Drude particles, and the virtual sites used in some water models to adjust the charge distribution. Extra particles can be recognized by the fact that their element is None.</p>
<p>This method is primarily used to add extra particles, but it can also remove them. It tries to match every residue in the Topology to a template in the force field. If there is no match, it will both add and remove extra particles as necessary to make it match.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">forcefield</td><td>(ForceField) the ForceField defining what extra particles should be present </td></tr>
  </table>
  </dd>
</dl>

<p>References <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a66aa7c8063db6217a0a0061f8b7ba206">Modeller.add()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmCrdFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmCrdFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ac17d0e3d420545b0cbe7c718cf026ddc">Modeller.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1desmonddmsfile_1_1DesmondDMSFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">DesmondDMSFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmRstFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmRstFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacsgrofile_1_1GromacsGroFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">GromacsGroFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a5e8a38af13069a0cc3fff9aae26892e4">Modeller.topology</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1amberprmtopfile_1_1AmberPrmtopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">AmberPrmtopFile.topology</a>, and <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacstopfile_1_1GromacsTopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">GromacsTopFile.topology</a>.</p>

</div>
</div>
<a class="anchor" id="ab19e4d48d3e7a317038fa32b82702ba6"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def addHydrogens </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>forcefield</em> = <code>None</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>pH</em> = <code>7.0</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>variants</em> = <code>None</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>platform</em> = <code>None</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Add missing hydrogens to the model. </p>
<p>Some residues can exist in multiple forms depending on the pH and properties of the local environment. These variants differ in the presence or absence of particular hydrogens. In particular, the following variants are supported:</p>
<p>Aspartic acid: ASH: Neutral form with a hydrogen on one of the delta oxygens ASP: Negatively charged form without a hydrogen on either delta oxygen</p>
<p>Cysteine: CYS: Neutral form with a hydrogen on the sulfur CYX: No hydrogen on the sulfur (either negatively charged, or part of a disulfide bond)</p>
<p>Glutamic acid: GLH: Neutral form with a hydrogen on one of the epsilon oxygens GLU: Negatively charged form without a hydrogen on either epsilon oxygen</p>
<p>Histidine: HID: Neutral form with a hydrogen on the ND1 atom HIE: Neutral form with a hydrogen on the NE2 atom HIP: Positively charged form with hydrogens on both ND1 and NE2</p>
<p>Lysine: LYN: Neutral form with two hydrogens on the zeta nitrogen LYS: Positively charged form with three hydrogens on the zeta nitrogen</p>
<p>The variant to use for each residue is determined by the following rules:</p>
<ol type="1">
<li>The most common variant at the specified pH is selected.</li>
<li>Any Cysteine that participates in a disulfide bond uses the CYX variant regardless of pH.</li>
<li>For a neutral Histidine residue, the HID or HIE variant is selected based on which one forms a better hydrogen bond.</li>
</ol>
<p>You can override these rules by explicitly specifying a variant for any residue. Also keep in mind that this function will only add hydrogens. It will never remove ones that are already present in the model, regardless of the specified pH.</p>
<p>Definitions for standard amino acids and nucleotides are built in. You can call <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a4d38bb19ab89760311eb99247c214948" title="Load an XML file containing definitions of hydrogens that should be added by addHydrogens(). ">loadHydrogenDefinitions()</a> to load additional definitions for other residue types.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">forcefield</td><td>(ForceField=None) the ForceField to use for determining the positions of hydrogens. If this is None, positions will be picked which are generally reasonable but not optimized for any particular ForceField. </td></tr>
    <tr><td class="paramname">pH</td><td>(float=7.0) the pH based on which to select variants </td></tr>
    <tr><td class="paramname">variants</td><td>(list=None) an optional list of variants to use. If this is specified, its length must equal the number of residues in the model. variants[i] is the name of the variant to use for residue i (indexed starting at 0). If an element is None, the standard rules will be followed to select a variant for that residue. </td></tr>
    <tr><td class="paramname">platform</td><td>(Platform=None) the Platform to use when computing the hydrogen atom positions. If this is None, the default Platform will be used. </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>a list of what variant was actually selected for each residue, in the same format as the variants parameter </dd></dl>

<p>References <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a97b90d5998ed9203e18087203e3eef2a">Modeller.getPositions()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmCrdFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmCrdFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ac17d0e3d420545b0cbe7c718cf026ddc">Modeller.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1desmonddmsfile_1_1DesmondDMSFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">DesmondDMSFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmRstFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmRstFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacsgrofile_1_1GromacsGroFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">GromacsGroFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a5e8a38af13069a0cc3fff9aae26892e4">Modeller.topology</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1amberprmtopfile_1_1AmberPrmtopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">AmberPrmtopFile.topology</a>, and <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacstopfile_1_1GromacsTopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">GromacsTopFile.topology</a>.</p>

</div>
</div>
<a class="anchor" id="ad489382394ab7f160e007ed890179435"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def addSolvent </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>forcefield</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>model</em> = <code>'tip3p'</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>boxSize</em> = <code>None</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>boxVectors</em> = <code>None</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>padding</em> = <code>None</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>numAdded</em> = <code>None</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>positiveIon</em> = <code>'Na+'</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>negativeIon</em> = <code>'Cl-'</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>ionicStrength</em> = <code>0*molar</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Add solvent (both water and ions) to the model to fill a rectangular box. </p>
<p>The algorithm works as follows:</p><ol type="1">
<li>Water molecules are added to fill the box.</li>
<li>Water molecules are removed if their distance to any solute atom is less than the sum of their van der Waals radii.</li>
<li>If the solute is charged, enough positive or negative ions are added to neutralize it. Each ion is added by randomly selecting a water molecule and replacing it with the ion.</li>
<li>Ion pairs are added to give the requested total ionic strength.</li>
</ol>
<p>The box size can be specified in any of several ways:</p>
<ol type="1">
<li>You can explicitly give the vectors defining the periodic box to use.</li>
<li>Alternatively, for a rectangular box you can simply give the dimensions of the unit cell.</li>
<li>You can give a padding distance. The largest dimension of the solute (along the x, y, or z axis) is determined, and a cubic box of size (largest dimension)+2*padding is used.</li>
<li>You can specify the total number of molecules (both waters and ions) to add. A cubic box is then created whose size is just large enough hold the specified amount of solvent.</li>
<li>Finally, if none of the above options is specified, the existing Topology's box vectors are used.</li>
</ol>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">forcefield</td><td>(ForceField) the ForceField to use for determining van der Waals radii and atomic charges </td></tr>
    <tr><td class="paramname">model</td><td>(string='tip3p') the water model to use. Supported values are 'tip3p', 'spce', 'tip4pew', and 'tip5p'. </td></tr>
    <tr><td class="paramname">boxSize</td><td>(Vec3=None) the size of the box to fill with water </td></tr>
    <tr><td class="paramname">boxVectors</td><td>(tuple of Vec3=None) the vectors defining the periodic box to fill with water </td></tr>
    <tr><td class="paramname">padding</td><td>(distance=None) the padding distance to use </td></tr>
    <tr><td class="paramname">numAdded</td><td>(int=None) the total number of molecules (waters and ions) to add </td></tr>
    <tr><td class="paramname">positiveIon</td><td>(string='Na+') the type of positive ion to add. Allowed values are 'Cs+', 'K+', 'Li+', 'Na+', and 'Rb+' </td></tr>
    <tr><td class="paramname">negativeIon</td><td>(string='Cl-') the type of negative ion to add. Allowed values are 'Cl-', 'Br-', 'F-', and 'I-'. Be aware that not all force fields support all ion types. </td></tr>
    <tr><td class="paramname">ionicStrength</td><td>(concentration=0*molar) the total concentration of ions (both positive and negative) to add. This does not include ions that are added to neutralize the system. </td></tr>
  </table>
  </dd>
</dl>

<p>References <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ac775ee34451fdfa742b318538164070e">Modeller.__init__()</a>, Modeller._ResidueData.hydrogens, Modeller._Hydrogen.maxph, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1element_1_1Element.html#a757840459670ee7692e00cf5ddc722d5">Element.name()</a>, ForceField._TemplateData.name, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1topology_1_1Residue.html#ab74e6bf80237ddc4109968cedc58c151">Residue.name</a>, ForceField._TemplateAtomData.name, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1topology_1_1Atom.html#ab74e6bf80237ddc4109968cedc58c151">Atom.name</a>, Modeller._ResidueData.name, Modeller._Hydrogen.name, Modeller._Hydrogen.parent, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmCrdFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmCrdFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ac17d0e3d420545b0cbe7c718cf026ddc">Modeller.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1desmonddmsfile_1_1DesmondDMSFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">DesmondDMSFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmRstFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmRstFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacsgrofile_1_1GromacsGroFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">GromacsGroFile.positions</a>, Modeller._Hydrogen.terminal, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a5e8a38af13069a0cc3fff9aae26892e4">Modeller.topology</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1amberprmtopfile_1_1AmberPrmtopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">AmberPrmtopFile.topology</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacstopfile_1_1GromacsTopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">GromacsTopFile.topology</a>, Modeller._ResidueData.variants, and Modeller._Hydrogen.variants.</p>

</div>
</div>
<a class="anchor" id="a6378af56ae75acce07d8a2f08f8cf87c"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def convertWater </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>model</em> = <code>'tip3p'</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Convert all water molecules to a different water model. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">model</td><td>(string='tip3p') the water model to convert to. Supported values are 'tip3p', 'spce', 'tip4pew', and 'tip5p'.</td></tr>
  </table>
  </dd>
</dl>
<dl class="deprecated"><dt><b><a class="el" href="deprecated.html#_deprecated000005">Deprecated:</a></b></dt><dd>Use <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ad59ef5b703f4dd7511858aa5d0be359d" title="Add missing extra particles to the model that are required by a force field. ">addExtraParticles()</a> instead. It performs the same function but in a more general way. </dd></dl>

<p>References <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmCrdFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmCrdFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ac17d0e3d420545b0cbe7c718cf026ddc">Modeller.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1desmonddmsfile_1_1DesmondDMSFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">DesmondDMSFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmRstFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmRstFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacsgrofile_1_1GromacsGroFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">GromacsGroFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a5e8a38af13069a0cc3fff9aae26892e4">Modeller.topology</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1amberprmtopfile_1_1AmberPrmtopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">AmberPrmtopFile.topology</a>, and <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacstopfile_1_1GromacsTopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">GromacsTopFile.topology</a>.</p>

</div>
</div>
<a class="anchor" id="ab48bde70a0927a83d62464cb0627c34b"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def delete </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>toDelete</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Delete chains, residues, atoms, and bonds from the model. </p>
<p>You can specify objects to delete at any granularity: atoms, residues, or chains. Passing in an Atom object causes that Atom to be deleted. Passing in a Residue object causes that Residue and all Atoms it contains to be deleted. Passing in a Chain object causes that Chain and all Residues and Atoms it contains to be deleted.</p>
<p>In all cases, when an Atom is deleted, any bonds it participates in are also deleted. You also can specify a bond (as a tuple of Atom objects) to delete just that bond without deleting the Atoms it connects.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">toDelete</td><td>(list) a list of Atoms, Residues, Chains, and bonds (specified as tuples of Atoms) to delete </td></tr>
  </table>
  </dd>
</dl>

<p>References <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmCrdFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmCrdFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ac17d0e3d420545b0cbe7c718cf026ddc">Modeller.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1desmonddmsfile_1_1DesmondDMSFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">DesmondDMSFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmRstFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmRstFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacsgrofile_1_1GromacsGroFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">GromacsGroFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a5e8a38af13069a0cc3fff9aae26892e4">Modeller.topology</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1amberprmtopfile_1_1AmberPrmtopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">AmberPrmtopFile.topology</a>, and <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacstopfile_1_1GromacsTopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">GromacsTopFile.topology</a>.</p>

<p>Referenced by <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#acb8683faf05f3bad2e7fd3985bf64963">Modeller.deleteWater()</a>.</p>

</div>
</div>
<a class="anchor" id="acb8683faf05f3bad2e7fd3985bf64963"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def deleteWater </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Delete all water molecules from the model. </p>

<p>References <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ab48bde70a0927a83d62464cb0627c34b">Modeller.delete()</a>.</p>

</div>
</div>
<a class="anchor" id="a97b90d5998ed9203e18087203e3eef2a"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def getPositions </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Get the atomic positions. </p>

<p>References <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmCrdFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmCrdFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ac17d0e3d420545b0cbe7c718cf026ddc">Modeller.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1desmonddmsfile_1_1DesmondDMSFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">DesmondDMSFile.positions</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1charmmcrdfiles_1_1CharmmRstFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">CharmmRstFile.positions</a>, and <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacsgrofile_1_1GromacsGroFile.html#ac17d0e3d420545b0cbe7c718cf026ddc">GromacsGroFile.positions</a>.</p>

<p>Referenced by <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ab19e4d48d3e7a317038fa32b82702ba6">Modeller.addHydrogens()</a>.</p>

</div>
</div>
<a class="anchor" id="a6bbbabdcb078f5feb40e195428ee93ee"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def getTopology </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Get the Topology of the model. </p>

<p>References <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a5e8a38af13069a0cc3fff9aae26892e4">Modeller.topology</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1amberprmtopfile_1_1AmberPrmtopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">AmberPrmtopFile.topology</a>, and <a class="el" href="classsimtk_1_1openmm_1_1app_1_1gromacstopfile_1_1GromacsTopFile.html#a5e8a38af13069a0cc3fff9aae26892e4">GromacsTopFile.topology</a>.</p>

</div>
</div>
<a class="anchor" id="a4d38bb19ab89760311eb99247c214948"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">def loadHydrogenDefinitions </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>file</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Load an XML file containing definitions of hydrogens that should be added by <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ab19e4d48d3e7a317038fa32b82702ba6" title="Add missing hydrogens to the model. ">addHydrogens()</a>. </p>
<p>The built in hydrogens.xml file containing definitions for standard amino acids and nucleotides is loaded automatically. This method can be used to load additional definitions for other residue types. They will then be used in subsequent calls to <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ab19e4d48d3e7a317038fa32b82702ba6" title="Add missing hydrogens to the model. ">addHydrogens()</a>. </p>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a class="anchor" id="ac17d0e3d420545b0cbe7c718cf026ddc"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">positions</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>The list of atom positions. </p>

<p>Referenced by <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a66aa7c8063db6217a0a0061f8b7ba206">Modeller.add()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ad59ef5b703f4dd7511858aa5d0be359d">Modeller.addExtraParticles()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ab19e4d48d3e7a317038fa32b82702ba6">Modeller.addHydrogens()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ad489382394ab7f160e007ed890179435">Modeller.addSolvent()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a6378af56ae75acce07d8a2f08f8cf87c">Modeller.convertWater()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ab48bde70a0927a83d62464cb0627c34b">Modeller.delete()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a97b90d5998ed9203e18087203e3eef2a">Modeller.getPositions()</a>, and <a class="el" href="classsimtk_1_1openmm_1_1app_1_1amberinpcrdfile_1_1AmberInpcrdFile.html#a97b90d5998ed9203e18087203e3eef2a">AmberInpcrdFile.getPositions()</a>.</p>

</div>
</div>
<a class="anchor" id="a5e8a38af13069a0cc3fff9aae26892e4"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">topology</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>The Topology describing the structure of the system. </p>

<p>Referenced by <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a66aa7c8063db6217a0a0061f8b7ba206">Modeller.add()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ad59ef5b703f4dd7511858aa5d0be359d">Modeller.addExtraParticles()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ab19e4d48d3e7a317038fa32b82702ba6">Modeller.addHydrogens()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ad489382394ab7f160e007ed890179435">Modeller.addSolvent()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a6378af56ae75acce07d8a2f08f8cf87c">Modeller.convertWater()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#ab48bde70a0927a83d62464cb0627c34b">Modeller.delete()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html#a6bbbabdcb078f5feb40e195428ee93ee">Modeller.getTopology()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1desmonddmsfile_1_1DesmondDMSFile.html#a6bbbabdcb078f5feb40e195428ee93ee">DesmondDMSFile.getTopology()</a>, <a class="el" href="classsimtk_1_1openmm_1_1app_1_1pdbfile_1_1PDBFile.html#a6bbbabdcb078f5feb40e195428ee93ee">PDBFile.getTopology()</a>, and <a class="el" href="classsimtk_1_1openmm_1_1app_1_1pdbxfile_1_1PDBxFile.html#a6bbbabdcb078f5feb40e195428ee93ee">PDBxFile.getTopology()</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>modeller.py</li>
</ul>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="navelem"><a class="el" href="namespacesimtk.html">simtk</a></li><li class="navelem"><a class="el" href="namespacesimtk_1_1openmm.html">openmm</a></li><li class="navelem"><a class="el" href="namespacesimtk_1_1openmm_1_1app.html">app</a></li><li class="navelem"><a class="el" href="namespacesimtk_1_1openmm_1_1app_1_1modeller.html">modeller</a></li><li class="navelem"><a class="el" href="classsimtk_1_1openmm_1_1app_1_1modeller_1_1Modeller.html">Modeller</a></li>
    <li class="footer">Generated on Thu Dec 17 2015 00:39:37 for OpenMM by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.8 </li>
  </ul>
</div>
</body>
</html>
